src/Backend/SynapseSearch.vala

```diff
@@ -116,6 +116,22 @@ namespace Slingshot.Backend {
             return pixbuf;
         }
 
+        public static async Gdk.Pixbuf? get_pathicon_for_match (Synapse.Match match, int size,
```

Comment:
This async method won't work asynchronously since you don't do anything asynchronous in it. You can remove it altogether.